Library/Homebrew/diagnostic.rb

```diff
@@ -443,6 +443,32 @@ def check_access_cellar
         EOS
       end
 
+      def check_for_empty_taps
+        taps_dir = HOMEBREW_LIBRARY/"Taps"
+        return unless taps_dir.exist?
+        return unless taps_dir.directory?
+
+        found_empty_dirs = []
+
+        # Warn USER of empty taps_dir children directories
+        taps_dir.children.select{ |c| c.directory? } \
```

Comment:
Unless I'm misunderstanding you, I think what you want is to detect empty directories that would be removed by `rmdir Library/Taps/* Library/Taps/*/*`. Is this correct? If so, here's a suggestion how to achieve this:

```ruby
taps_dir = HOMEBREW_LIBRARY/"Taps"
return unless taps_dir.directory?

user_dirs = taps_dir.children.select { |pn| !pn.symlink? && pn.directory? }
repo_dirs = user_dirs.flat_map(&:children).select { |pn| !pn.symlink? && pn.directory? }
empty_dirs = (user_dirs + repo_dirs).select { |pn| pn.children.empty? }
return if empty_dirs.empty?
```

This will additionally make sure to skip over any symbolic links, to avoid deleting them and recursing into them (I think that's desirable).